### ISSUE TITLE
terraform apply only idempotent on second run

### ISSUE THREAD

**Issue by gschlossnagle** [hashicorp/terraform#1664]
First run:

```
Georges-MacBook-Air:matrix george$ terraform apply -target=aws_vpc_peering_connection.openvpn -target=aws_vpc_peering_connection.spe
aws_vpc.default: Creating...
  cidr_block:                "" => "10.99.160.0/21"
  default_network_acl_id:    "" => "<computed>"
  default_security_group_id: "" => "<computed>"
  enable_dns_hostnames:      "" => "1"
  enable_dns_support:        "" => "<computed>"
  main_route_table_id:       "" => "<computed>"
  tags.#:                    "" => "1"
  tags.Name:                 "" => "vpc-matrix-usw2-prd"
aws_vpc.default: Creation complete
aws_vpc_peering_connection.openvpn: Creating...
  peer_owner_id: "" => "266591500818"
  peer_vpc_id:   "" => "vpc-3642fa53"
  tags.#:        "" => "1"
  tags.Name:     "" => "openvpn-spe-matrix-usw2-pcx"
  vpc_id:        "" => "vpc-bb8d09de"
aws_vpc_peering_connection.spe: Creating...
  peer_owner_id: "" => "266591500818"
  peer_vpc_id:   "" => "vpc-2fe3124a"
  tags.#:        "" => "1"
  tags.Name:     "" => "pcx-admin_prd-matrix_prd-usw2"
  vpc_id:        "" => "vpc-bb8d09de"
aws_vpc_peering_connection.spe: Creation complete
aws_vpc_peering_connection.openvpn: Creation complete

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
```

Second run:

```
Georges-MacBook-Air:matrix george$ terraform plan -target=aws_vpc_peering_connection.openvpn -target=aws_vpc_peering_connection.spe
Refreshing Terraform state prior to plan...

aws_vpc.default: Refreshing state... (ID: vpc-bb8d09de)
aws_vpc_peering_connection.openvpn: Refreshing state... (ID: pcx-840bf6ed)
aws_vpc_peering_connection.spe: Refreshing state... (ID: pcx-850bf6ec)

The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

~ aws_vpc_peering_connection.openvpn
    tags.#:    "0" => "1"
    tags.Name: "" => "openvpn-spe-matrix-usw2-pcx"

~ aws_vpc_peering_connection.spe
    tags.#:    "0" => "1"
    tags.Name: "" => "pcx-admin_prd-matrix_prd-usw2"


Georges-MacBook-Air:matrix george$ terraform apply -target=aws_vpc_peering_connection.openvpn -target=aws_vpc_peering_connection.spe
aws_vpc.default: Refreshing state... (ID: vpc-bb8d09de)
aws_vpc_peering_connection.spe: Refreshing state... (ID: pcx-850bf6ec)
aws_vpc_peering_connection.openvpn: Refreshing state... (ID: pcx-840bf6ed)
aws_vpc_peering_connection.spe: Modifying...
  tags.#:    "0" => "1"
  tags.Name: "" => "pcx-admin_prd-matrix_prd-usw2"
aws_vpc_peering_connection.openvpn: Modifying...
  tags.#:    "0" => "1"
  tags.Name: "" => "openvpn-spe-matrix-usw2-pcx"
aws_vpc_peering_connection.spe: Modifications complete
aws_vpc_peering_connection.openvpn: Modifications complete

Apply complete! Resources: 0 added, 2 changed, 0 destroyed.
```

Third run:

```
Georges-MacBook-Air:matrix george$ terraform plan -target=aws_vpc_peering_connection.openvpn -target=aws_vpc_peering_connection.spe
Refreshing Terraform state prior to plan...

aws_vpc.default: Refreshing state... (ID: vpc-bb8d09de)
aws_vpc_peering_connection.spe: Refreshing state... (ID: pcx-850bf6ec)
aws_vpc_peering_connection.openvpn: Refreshing state... (ID: pcx-840bf6ed)

No changes. Infrastructure is up-to-date. This means that Terraform
could not detect any differences between your configuration and
the real physical resources that exist. As a result, Terraform
doesn't need to do anything.
Georges-MacBook-Air:matrix george$ terraform apply -target=aws_vpc_peering_connection.openvpn -target=aws_vpc_peering_connection.spe
aws_vpc.default: Refreshing state... (ID: vpc-bb8d09de)
aws_vpc_peering_connection.openvpn: Refreshing state... (ID: pcx-840bf6ed)
aws_vpc_peering_connection.spe: Refreshing state... (ID: pcx-850bf6ec)
```

Why don't the tags get associated on the first run?

**Comment by johnrengelman** [comment]
I can confirm this one as well. VPC Peering Connection Tags are not applied on resource creation.

**Comment by johnrengelman** [comment]
This should fix this problem: https://github.com/hashicorp/terraform/pull/1672

**Comment by phinze** [comment]
Thanks for the report @gschlossnagle and the fix @johnrengelman - just merged so this should be fixed in master and the upcoming 0.5.0.
